lib/widgets/subscription_list.dart

```diff
@@ -239,6 +243,10 @@ class SubscriptionItem extends StatelessWidget {
             // TODO(#384) show @-mention indicator when it applies
             UnreadCountBadge(count: unreadCount, backgroundColor: swatch, bold: true),
           ],
+          if (hasMutedUnreads) ...[
```

Comment:
```suggestion
          ] else ...[
```
Can't we write something like this? 
And I wonder how is `hasMutedUnreads` different from `unreadCount <= 0`?!

Also upon interacting with the web, I noticed three interesting cases that control how the `SubscriptionItem` is shown:

1. The stream is unmuted, so everything looks normal with the unread count badge present.
2. The stream is muted and all its topics are in the **Default** state, so the icon and stream title have an opacity and the unread count badge is replaced by a gray color dot.
3. The stream is muted but one or more of its topics are in the **Unmute** state, so the icon and stream title have an opacity and there is again the unread count badge showing the sum of the **Unmute** topics' unread count. (This is the reason I suggested making the `Opacity` control only the icon and the stream title in [711 comment](https://github.com/zulip/zulip-flutter/pull/711#discussion_r1622478616).)

So to match it with the web, I think we have to consider these three cases.

Reply:
- `unreadCount > 0`: there are unreads that are not unmuted.
- `hasMutedUnreads`: there are unreads that are muted.
The three states that we have are:

| unreadCount > 0 | hasMutedUnreads | badge |
|---|---|---|
| true | any | UnreadCountBadge |
| false | true | MutedUnreadBadge |
| false | false | None | 

So you can see that we don't need to calculate `hasMutedUnreads` in case of `unreadCount > 0` being true, which could save some heavy computing in some cases. So I try to avoid any unnecessary computations by having the `hasMutedUnreads` being conditioned on `unreadCount == 0`.
```dart
        final hasMutedUnreads = unreadCount == 0 && unreadsModel!.hasMutedInStream(subscription.streamId);
```
which would cause as a side effect that guarantees that:
```dart
 unreadCount > 0 && hasMutedUnreads = false // this is guaranteed always
 ```
 So finally I'd update the code for good measure as:
 ```dart
          if (unreadCount > 0) ...[
            const SizedBox(width: 12),
            // TODO(#384) show @-mention indicator when it applies
            UnreadCountBadge(count: unreadCount, backgroundColor: swatch, bold: true),
          ] else if (hasMutedUnreads) ...[ // else here is not required but it is best to have it just to make sure
            const SizedBox(width: 12),
            const MutedUnreadBadge(),
          ],
```

Reply:
I am snapping my head around to understand the difference, but unfortunately, I am not able to 😀. Specifically, as you said:
> unreadCount > 0: there are unreads that are not unmuted.

vs.
> hasMutedUnreads: there are unreads that are muted.

Isn't **_not unmuted_** = **_muted_**?

Also while interacting with the UI, I couldn't notice any difference between these changes and when I replace `else if (hasMutedUnreads)` with `else` and completely ignore the `hasMutedUnreads`!

Reply:
Oh sorry... Actually there is a mistake here
`unreadCount > 0`: there are unreads that are ~~not~~ unmuted.